### DISCUSSION TITLE
Feature Request: Strict Path Restrictions for Different Vehicle Types in OpenTCS

### DISCUSSION THREAD

**Discussion by csliao0524** [openTCS/opentcs#148] [Q&A] [answered]
Hi, 

As far as I know, OpenTCS's default router has the capability to calculate different costs for different types of vehicles to plan suitable routes, as demonstrated in the discussion titled "Resource release problem in OpenTCS v5.16 #145."
 
However, if only the cost is used to regulate whether a vehicle can or cannot travel on a certain route, there is a possibility that when all other routes are locked, a vehicle might still take routes with high costs that we originally did not want it to travel on. This could lead to safety issues in practical applications (e.g., a large vehicle forcing its way through a narrow passage).
 
Therefore, my question is: Does OpenTCS currently have the functionality to strictly limit different types of vehicles to travel only on specified paths? If not, is there a plan to include this feature in future development?
 
Note:
The only related content I found is in section 4.2.3 if OpenTCS user guide "Avoiding/Excluding resource when computing routes," which uses the API to include properties like tcs:resourcesToAvoid to actively restrict specific vehicles (not by vehicle type).

**Comment by swltr** [comment, marked as answer]
@csliao0524 When using property-based costs for paths (see https://opentcs.org/docs/6.0/user/opentcs-users-guide.html#_cost_functions), you can set the property value to the string `"Infinity"` to express that a path cannot be taken.
